### PR TITLE
fix(edgeless): paste as note block if all other cases fails

### LINKED ISSUE #6609: SyntaxError: "undefined" is not valid JSON
Sentry Issue: [AFFINE-513](https://toeverything.sentry.io/issues/5109431971/?referrer=github_integration)

```
SyntaxError: "undefined" is not valid JSON
  at JSON.parse (<anonymous>)
  at Set.forEach (<anonymous>)
...
(7 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by golok727**
Close https://github.com/toeverything/blocksuite/issues/6788, close #6609.

https://github.com/toeverything/blocksuite/assets/123532141/dabb1f6e-245e-4649-85df-db0f3adf9237

### CODE REVIEW

**Comment by donteatfriedrice** [comment]
Thank you for your work. We plan to gradually reduce the use of canvas text in the future. Instead of pasting the content into canvas text, it may be a better way to add a note block at the paste position and paste the content into the newly added note block. Are you willing to do this? Thanks again!

**Comment by golok727** [comment]
@donteatfriedrice PTAL.

https://github.com/toeverything/blocksuite/assets/123532141/dabb1f6e-245e-4649-85df-db0f3adf9237

File: packages/blocks/src/surface-block/elements/text/utils.ts [hunk 1]
```diff
@@ -102,19 +102,19 @@ export function getTextRect(
...
+    .map(x => x || ' ')
+    .join('\n');
+
+  const lineHeight = getLineHeight(
+    fontFamily as CanvasTextFontFamilyValueType,
+    fontSize
+  );
   const font = `${fontSize}px "${fontFamily}"`;
-
-  if (font !== ctx.font) ctx.font = font;
```
Comment (fourdim) [inline packages/blocks/src/surface-block/elements/text/utils.ts:108]: Why are you removing this line?

**Review by fourdim — APPROVED** [review]
LGTM.

**Pull request merged**